File: python/blockchain.py
from typing import List
from block import Block


class Blockchain():
    def __init__(self) -> None:
        self.blocks: List[Block] = []

    def add_block(self, block: Block):
        self.blocks.append(block)

    def print(self):
        for blk in self.blocks:
            print(blk)


if __name__ == "__main__":
    chain: Blockchain = Blockchain()
    # add a dummy block as the header, this is called genesis block
    gblk = Block("genesis", "0")
    gblk.mine()
    chain.add_block(gblk)
    # add 5 blocks
    prev_blk = gblk
    data_items = ["test1", "test2", "test3", "test4", "test5"]
    difficulty = 5
    for data in data_items:
        blk: Block = Block(data, prev_blk.hash, difficulty)
        blk.mine()
        chain.add_block(blk)
        prev_blk = blk
    chain.print()

File: nodejs/blockchain.js
const Block = require('./block')


class Blockchain {
    constructor() {
        this.blocks = []
    }

    addBlock = (block) => {
        this.blocks.push(block)
    }

    print = () => {
        this.blocks.forEach(blk => {
            console.log(blk.toString())
        })
    }
}

if (require.main === module) {
    var chain = new Blockchain()
    // add a dummy block as the header, this is called genesis block
    var gblk = new Block("genesis", "0")
    gblk.generateHash()
    chain.addBlock(gblk)
    // add 5 blocks
    var prevBlk = gblk
    var dataList = ["test1", "test2", "test3", "test4", "test5"]
    dataList.forEach(data => {
        var blk = new Block(data, prevBlk.hash)
        blk.generateHash()
        chain.addBlock(blk)
        prevBlk = blk
    })
    chain.print()
}
